src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -45,6 +47,12 @@ public Envelope processEnvelope(byte[] metadataStream, String containerName) thr
         Envelope envelope = EntityParser.parseEnvelopeMetadata(inputStream);
         envelope.setContainer(containerName);
 
+        envelopeRepository
+            .findByContainerAndZipFileName(containerName, envelope.getZipFileName())
+            .ifPresent(dbEnvelope -> {
+                throw new DuplicateEnvelopeException(dbEnvelope);
```

Comment:
Rather passing envelope we could pass a message like 
```
throw new DuplicateEnvelopeException("Envelope already exists with zip file name" +  envelope.getZipFileName() + "and status" + envelope.getStatus() );
```

Reply:
Envelope is used in `catch` handler to update status and record event